Updated API Fingerprint and switched to a BaseUrl variable for API location

kcranky opened the pull request:
Updated the footprint (fingerprint?) and gave the new date with month text to prevent date ambiguity.

Added a BaseURL for specifying API root address. Checking the old problematic ESP8266 method (const String Toggl::getProject(int const& WID)) works correctly with the new base URL.

kcranky committed:
Updated API Fingerprint and switched to a BaseUrl variable for API location

JoeyStrandnes commented:
Thanks for the contribution, will include it in the next release!

kcranky commented:
You're welcome! There's one or two other things I think could be improved on (for example not returning key/value pairs as strings) - would you prefer I create new issues in the primary repo to fully elaborate on these, or can I go ahead and make a new PR branching off of this?

JoeyStrandnes commented:
Not sure what happened with your question regarding "String" return type and other improvements.

I think it would make it easier to follow the development cycle if you could create a new issue to elaborate on the improvements.

kcranky commented:
Thanks, figured it was best to do a new issue, so I removed it. Wanted to do some testing first so that's why the issue didn't go up straight away. I've added it: https://github.com/JoeyStrandnes/Arduino-Toggl-API/issues/7
I'll add more issues after hours for any other things I think are worth raising. Thanks for being open to discussing these things!
